ccl/multiregionccl: TestRegionLivenessProberForLeases failed

ccl/multiregionccl.TestRegionLivenessProberForLeases [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15543486?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15543486?buildTab=artifacts#/) on master @ [b489a1d35492e4631b7f0f7856767a1c7f094820](https://github.com/cockroachdb/cockroach/commits/b489a1d35492e4631b7f0f7856767a1c7f094820):


```
=== RUN   TestRegionLivenessProberForLeases
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/6c69b7baea525e4f03f56a8383206787/logTestRegionLivenessProberForLeases3933364360
    test_log_scope.go:81: use -show-logs to present logs inline
    regionliveness_test.go:446: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test/pkg/ccl/multiregionccl/regionliveness_test.go:446
        	Error:      	Error "pq: Could not complete operation in a failed transaction" does not contain "TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn"
        	Test:       	TestRegionLivenessProberForLeases
        	Messages:   	connection should have been dropped, node is dead.
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/6c69b7baea525e4f03f56a8383206787/logTestRegionLivenessProberForLeases3933364360
--- FAIL: TestRegionLivenessProberForLeases (90.16s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionLivenessProberForLeases.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39325

ccl/multiregionccl.TestRegionLivenessProberForLeases [failed](https://tanzanite.cluster.engflow.com/invocations/default/58f54c5f-5bad-45b6-8d5b-691f313555ef?testReportRun=4&testReportShard=13&testReportAttempt=1#targets-Ly9wa2cvY2NsL211bHRpcmVnaW9uY2NsOm11bHRpcmVnaW9uY2NsX3Rlc3Q=) on master @ [802554ac59763c31aee034651efcd28f7bdcdd5d](https://github.com/cockroachdb/cockroach/commits/802554ac59763c31aee034651efcd28f7bdcdd5d):


```
=== RUN   TestRegionLivenessProberForLeases
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestRegionLivenessProberForLeases1056993635
    test_log_scope.go:81: use -show-logs to present logs inline
    regionliveness_test.go:446: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test/pkg/ccl/multiregionccl/regionliveness_test.go:446
        	Error:      	Error "pq: Could not complete operation in a failed transaction" does not contain "TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn"
        	Test:       	TestRegionLivenessProberForLeases
        	Messages:   	connection should have been dropped, node is dead.
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestRegionLivenessProberForLeases1056993635
--- FAIL: TestRegionLivenessProberForLeases (86.90s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=13</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionLivenessProberForLeases.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/multiregionccl.TestRegionLivenessProberForLeases [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15608844?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15608844?buildTab=artifacts#/) on master @ [ed40d3c996fae6efd323d94e8982d95732dc6e4e](https://github.com/cockroachdb/cockroach/commits/ed40d3c996fae6efd323d94e8982d95732dc6e4e):


```
=== RUN   TestRegionLivenessProberForLeases
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/6c69b7baea525e4f03f56a8383206787/logTestRegionLivenessProberForLeases3347501766
    test_log_scope.go:81: use -show-logs to present logs inline
    regionliveness_test.go:446: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test/pkg/ccl/multiregionccl/regionliveness_test.go:446
        	Error:      	Error "pq: Could not complete operation in a failed transaction" does not contain "TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn"
        	Test:       	TestRegionLivenessProberForLeases
        	Messages:   	connection should have been dropped, node is dead.
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/6c69b7baea525e4f03f56a8383206787/logTestRegionLivenessProberForLeases3347501766
--- FAIL: TestRegionLivenessProberForLeases (87.17s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionLivenessProberForLeases.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/multiregionccl.TestRegionLivenessProberForLeases [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15617309?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15617309?buildTab=artifacts#/) on master @ [00d41392f1deb1d87e5b9e541f660ec8d04ccbb7](https://github.com/cockroachdb/cockroach/commits/00d41392f1deb1d87e5b9e541f660ec8d04ccbb7):


```
=== RUN   TestRegionLivenessProberForLeases
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/6c69b7baea525e4f03f56a8383206787/logTestRegionLivenessProberForLeases2478361498
    test_log_scope.go:81: use -show-logs to present logs inline
    regionliveness_test.go:446: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test/pkg/ccl/multiregionccl/regionliveness_test.go:446
        	Error:      	Error "pq: Could not complete operation in a failed transaction" does not contain "TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn"
        	Test:       	TestRegionLivenessProberForLeases
        	Messages:   	connection should have been dropped, node is dead.
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/6c69b7baea525e4f03f56a8383206787/logTestRegionLivenessProberForLeases2478361498
--- FAIL: TestRegionLivenessProberForLeases (90.06s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionLivenessProberForLeases.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/multiregionccl.TestRegionLivenessProberForLeases [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15632739?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15632739?buildTab=artifacts#/) on master @ [718b6cad0408b5d785449f2e2a189f1a8e6ff1ae](https://github.com/cockroachdb/cockroach/commits/718b6cad0408b5d785449f2e2a189f1a8e6ff1ae):


```
=== RUN   TestRegionLivenessProberForLeases
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/6c69b7baea525e4f03f56a8383206787/logTestRegionLivenessProberForLeases243852435
    test_log_scope.go:81: use -show-logs to present logs inline
    regionliveness_test.go:446: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test/pkg/ccl/multiregionccl/regionliveness_test.go:446
        	Error:      	Error "pq: Could not complete operation in a failed transaction" does not contain "TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn"
        	Test:       	TestRegionLivenessProberForLeases
        	Messages:   	connection should have been dropped, node is dead.
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/6c69b7baea525e4f03f56a8383206787/logTestRegionLivenessProberForLeases243852435
--- FAIL: TestRegionLivenessProberForLeases (92.39s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionLivenessProberForLeases.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/ccl/multiregionccl/regionliveness_test.go
+++ b/pkg/ccl/multiregionccl/regionliveness_test.go
@@ -442,8 +442,21 @@ func TestRegionLivenessProberForLeases(t *testing.T) {
 	recoveryBlock <- struct{}{}
 	require.NoError(t, grp.Wait())
 	_, err = tx.Exec("INSERT INTO t2 VALUES(5)")
+	// If the txn failed, no commit is needed.
+	const expectedTxnErr = "restart transaction: TransactionRetryWithProtoRefreshError"
+	if err != nil {
+		require.ErrorContainsf(t,
+			err,
+			expectedTxnErr,
+			"txn should see a retry error.")
+		require.NoError(t, tx.Rollback())
+	} else {
+		require.ErrorContainsf(t,
+			tx.Commit(),
+			expectedTxnErr,
+			"txn should see a retry error")
+	}
 	require.ErrorContainsf(t, grp.Wait(), "context canceled", "connection should have been dropped, node is dead.")
-	require.ErrorContainsf(t, tx.Commit(), "TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn", "connection should have been dropped, node is dead.")
 }
 
 // TestRegionLivenessProberForSQLInstances validates that regional avaibility issues

